src/js/references/components/ReferenceForm.tsx

```diff
@@ -0,0 +1,63 @@
+import React from "react";
+import { FieldErrors, UseFormRegister } from "react-hook-form";
+import styled from "styled-components";
+import { InputError, InputGroup, InputLabel, InputSimple, TextArea } from "../../base";
+import { ReferenceDataType } from "../types";
+
+const StyledInputGroup = styled(InputGroup)`
+    padding-bottom: 0px;
+`;
+
+export enum Mode {
+    edit = "edit",
+    empty = "empty",
+}
+
+type FormValues = {
+    name: string;
+    description: string;
+    dataType: ReferenceDataType;
+    organism: string;
+};
+
+type ReferenceFormProps = {
+    /** Form validation errors */
+    errors: FieldErrors<FormValues>;
+    /** The mode of the reference form */
+    mode: Mode;
+    /** Function to register form fields */
+    register: UseFormRegister<FormValues>;
+};
+
+/**
+ * Form input fields for organism, name and description
+ */
+export function ReferenceForm({ errors, mode, register }: ReferenceFormProps) {
+    let organismComponent;
+
+    if (mode === "empty" || mode === "edit") {
```

Comment:
Since your defining an enum you can do the comparison with the enum rather than the raw string